Feature/setup-environment branch

Sam09040 opened the pull request:
- create a test environment
- connect to testdb
- run tests creating an user and removing them after the test has ended

Sam09040 committed:
Inicial setup

- test file with user mutation in testdb functioning

- add app folder and put interfaces, client and graphql content there

Sam09040 committed:
Small changes

- update prisma version
- correct index.test.mjs

racsandoval reviewed (commented):
esse PR está pegando arquivos do PR anterior

Sam09040 commented:
> esse PR está pegando arquivos do PR anterior

sim, eu usei o cherry picking para continuar de onde parei, mas eu não dei commit no cherry pick para não alterar muito o histórico. isso está errado? eu deveria ter dado commit nos cherry picks?

racsandoval commented:
> > esse PR está pegando arquivos do PR anterior
> 
> sim, eu usei o cherry picking para continuar de onde parei, mas eu não dei commit no cherry pick para não alterar muito o histórico. isso está errado? eu deveria ter dado commit nos cherry picks?

Uma boa abordagem seria apontar o PR para a branch de que ele saiu, assim só apareceriam os arquivos realmente alterados nessa branch. E no momento do merge poderia mudar de novo para `develop`

Sam09040 committed:
small changes

- tsimp to run .ts
- run configuration with tsimp

- test file from .mjs to .ts
- resolvers import on index.ts

- .prettierrc.ts file as it was a duplicate

Sam09040 committed:
correction

## remove
- test string from resolvers and schema

racsandoval commented on .prettierrc.js:
```diff
@@ -1,8 +1,9 @@
-module.exports = {
+const env = {
```
pq chamou essa const de `env`?

Sam09040 replied:
precisava de um nome para a variável e eu escolhi env de environment porque achei cabível já que faz parte de configurações do projeto. Se for mais adequado, posso chamar de `config` ou outro nome que não env.

racsandoval commented on test/index.test.ts:
```diff
@@ -0,0 +1,90 @@
+/* eslint-disable no-undef */
```
pq precisou remover essa regra?

Sam09040 replied:
honestamente, não me lembro o motivo desse erro estar acontecendo, mas acredito que consegui resolver e não precisei mais dela.

Sam09040 replied:
me lembrei, eu acho! antes, por algum motivo, os métodos do mocha, como describe e it, davam como não definidos, então precisei remover a regra. porém, após uma mudança no tsconfig, acredito eu que adicionar "mocha" a "types", não foi mais necessário essa regra.

racsandoval commented on src/app/graphql/resolvers.ts:
```diff
@@ -0,0 +1,75 @@
+    users: async () => {
+      const users = await prisma.user.findMany();
+
+      if (!users) {
+        return null;
+      }
```
pq aqui retorna null? não poderia ser uma array vazia?

Sam09040 replied:
acho que isso ocorreu mais por conta de costume da maneira que eu trato esse tipo de situação, onde uma variável é nula ou vazia, logo retorno `null`

racsandoval commented on test/index.test.ts:
```diff
@@ -0,0 +1,90 @@
+    
+    before(async () => {
+        serverInstance = server;
+        if(!server.listen()){
+            server.listen(port).then( async ({ url }) => {
+                console.log(url);
+            })       
+        }
```
achei um pouco confuso aqui. Aqui está executando a função duas vezes, uma vez dentro da condição do if sem a porta e outra dentro do if com a parta. Pq o `server.listen` é executado duas vezes?

Sam09040 replied:
o que está no if é para verificar se o servidor está conectado, enquanto o segundo é para conectar, caso não esteja conectado.

Sam09040 replied:
eu percebo, agora, que foi uma maneira muito suja e confusa de tratar isso.

racsandoval commented on test/index.test.ts:
```diff
@@ -0,0 +1,90 @@
+                console.log(url);
+            })       
+        }
+        if(!prisma.$connect()){
+            await prisma.$connect();
+            console.log(`Connected to database testdb`);
+        }
```
mesma coisa aqui

Sam09040 replied:
mesma coisa, o primeiro é para verificar se está conectado e o segundo para conectar, caso não esteja conectado. vou corrigir esses erros no próximo commit.

racsandoval commented on test/index.test.ts:
```diff
@@ -0,0 +1,90 @@
+        }
+    });
+
+    it('should create an user successfully', async () => {
```
poderia testar os erros da chamada também

racsandoval replied:
vi que está no próximo PR

Sam09040 committed:
corrections

- changed prettier export variable name to 'prettier';
- removed eslint.config.mjs;
- connection verification on before all hook.

walteraandrade commented on .gitignore:
```diff
@@ -3,3 +3,5 @@
 /db
 *.env*
 /prisma/migrations
+.vscode
```
parece que esses commits, do pr anterior, ainda estão aqui 🤔 
parece bobo, mas acho bom investigar o que está acontecendo, em casos de projeto real isso pode causar problemas maiores

Sam09040 replied:
na real, não são o mesmo commit, mas eu fiz coisas parecidas nas duas instâncias. olhando o período, a mensagem, o nome, entre outros, só foram mudanças parecidas.

walteraandrade commented on src/app/graphql/resolvers.ts:
```diff
@@ -0,0 +1,75 @@
+
+      return users;
+    },
+    user: async (_: unknown, { id }: { id: number }) => {
```
por que esses argumentos que não são utilizados?

Sam09040 replied:
o `_: unknown` refere-se a parte parent do resolver, onde retorna o "pai" daquele campo, ou então o rootValue para campos que não tenham "pai". ele é necessário para o resolver identificar o que é o que, então o resolver sabe que essa primeira parte é referente ao parent, enquanto a parte do `id` são os argumentos que pode receber.

walteraandrade commented on src/index.ts:
```diff
@@ -1,40 +1,16 @@
-});
+export default server;
+
+/* server.listen(process.env.PORT).then(async ({ url }) => {
```
essas mudanças parecem com as do pr anterior também

Sam09040 replied:
são um pouco parecidas mesmo, mas são commits diferentes.

walteraandrade commented on test/index.test.ts:
```diff
@@ -0,0 +1,88 @@
+            variables,
+        });
+        const { data } = response.data;
+        console.log('response received: ', data);
```
acho que os logs são úteis pra ajudar no desenvolvimento, mas buscamos evitar comentá-los. se você já está usando o `expect`  pra validar a resposta, esse logs não fazem muita coisa

Sam09040 replied:
okay!

Sam09040 committed:
fixes after rebase

## updated:
- imports

## removed:
- unnecessary logs

sophia-bergamo commented on src/app/graphql/resolvers.ts:
```diff
@@ -1,18 +1,16 @@
     users: async () => {
       const users = await prisma.user.findMany();
 
       if (!users) {
```
```suggestion
      if (!users.length) {
```

Sam09040 committed:
small fix

## update:
- change users verification from `!users` to `!users.lenght`
